src/compositor/input/pointer.rs

```diff
@@ -40,35 +48,52 @@ impl PointerHandler for Pointer {
                          ref mut seat,
                          ref mut keyboards,
                          .. } = *server;
-            if event.state() == WLR_BUTTON_RELEASED {
-                seat.action = None;
-                send_pointer_button(seat, event).expect("Could not send pointer button");
-                return
-            }
-            if let Some(view) = view_at_pointer(views, cursor) {
-                focus_under_pointer(seat, &mut **keyboards, { &mut *view }).expect("Could not focus \
+            with_handles!([(cursor: {&mut *cursor})] => {
+                if event.state() == WLR_BUTTON_RELEASED {
+                    seat.action = None;
+                    send_pointer_button(seat, event).expect("Could not send pointer button");
+                    return
+                }
+                if let Some(view) = view_at_pointer(views, cursor) {
+                    focus_under_pointer(seat, &mut **keyboards, { &mut *view }).expect("Could not focus \
                                                                                     view");
+                    let meta_held_down = seat.meta;
+                    if meta_held_down && event.button() == BTN_LEFT {
+                        move_view(seat, cursor, view, None);
+                    }
+                    send_pointer_button(seat, event).expect("Could not send pointer button");
+                } else {
+                    focus_under_pointer(seat, &mut **keyboards, None).expect("Could not focus view");
+                }
+            }).unwrap()
+        }).unwrap()
+    }
+}
+
+/// After the cursor has been warped, do all the things needed to change the cursor from the old
+/// position to the new position.
+fn update_cursor_position(cursor: &mut Cursor, seat: &mut Seat, views: &mut [View]) {
```

Comment:
This doesn't actually update the cursor position though. It updates the views to be where the cursor is we should move the window (which in this proto-way-cooler phase it always is). If the idea is it should do other generic stuff then I think we shouldn't try to shoe horn those things into into this function and just focus on updating views.

This function name and doc comment should be changed to reflect that. (e.g. `move_focused_views_to_cursor` or something to that effect).

Reply:
It will need to do other stuff too though like the resize.

Reply:
Ok, but that isn't really updating the cursor position then is it? It should be `update_view_position`

Reply:
We can call it that for now and then reevaluate the name later once you see the stuff that's going to go in there.